keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -180,7 +186,7 @@ public long getId() {
     }
 
     public Set<Vertex> getChildren() {
-        return children;
+        return ImmutableSet.copyOf(children);
```

Comment:
Creating these copies of children and parents will happen a lot. Could we store it has an immutable collection and return that? And then if we ever need to change the children or parents we could swap out the entire collection.

Reply:
That sounds reasonable.